src/tasks/test-deployment.ts

```diff
@@ -35,7 +35,7 @@ const OUTPUT_FOLDER = "./output/test-deployment";
 
 const defaultArgs = {
   userCount: 1000,
-  totalSupply: (10n ** (3n * 4n)).toString(),
+  totalSupply: (10n ** (3n * 3n)).toString(),
```

Comment:
Nit: I was a little bit confused, as sometimes decimals are counted and sometimes they are not. Further below you are writing:
```
    totalSupply: BigNumber.from(10).pow(3 * 3 + metadata.real.decimals),
    ```
    Maybe you can use everywhere:
    
    ```
    utils.parseUnits(
      BigNumber.from(10).pow(3 * 3 )
      metadata.real.decimals,
    ),
    ```